modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheDeploymentManager.java

```diff
@@ -947,7 +947,8 @@ private boolean isLocallyExcluded(String name) {
          */
         private CachedDeploymentInfo(UUID sndId, IgniteUuid ldrId, String userVer, DeploymentMode depMode,
             Map<UUID, IgniteUuid> participants) {
-            assert sndId.equals(ldrId.globalId()) || participants != null;
+            assert sndId.equals(ldrId.globalId()) || participants != null : "[senderId=" + sndId +
```

Comment:
I would prefer to add a small message before detailed info as follows:
`Unable to create the deployment info [senderId....]`